smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/NamespaceFetcher.java

```diff
@@ -143,28 +181,47 @@ public void run() {
         }
         return;
       }
+      String configString = conf.get(SmartConfKeys.SMART_SERVER_IGNORE_FILE_DIRS_KEY);
+      if (configString == null){
+        configString = "";
+      }
+
+      //only when parent dir is not ignored we run the follow code
+      List<String> ignoreList = Arrays.asList(configString.split(","));
+
+      boolean isIgnored = false;
+      System.out.println(ignoreList);
+      for (int i = 0; i < ignoreList.size(); i++) {
+        if (ignoreList.get(i).equals(parent)) {
+          isIgnored = true;
+          break;
+        }
+      }
 
-      try {
-        HdfsFileStatus status = client.getFileInfo(parent);
-        if (status != null && status.isDir()) {
-          FileInfo internal = convertToFileInfo(status,"");
-          internal.setPath(parent);
-          this.addFileStatus(internal);
-          numDirectoriesFetched++;
-          HdfsFileStatus[] children = this.listStatus(parent);
-          for (HdfsFileStatus child : children) {
-            if (child.isDir()) {
-              this.deque.add(child.getFullName(parent));
-            } else {
-              this.addFileStatus(convertToFileInfo(child, parent));
-              numFilesFetched++;
+      if (!isIgnored) {
```

Comment:
Just return when isIgnored = true then no changes will be made to the following lines.